File: Back-End/src/main/java/fr/ensicaen/pi/gpss/backend/database/dto/account/BankAccountDTO.java
package fr.ensicaen.pi.gpss.backend.database.dto.account;

import com.fasterxml.jackson.annotation.JsonInclude;
import fr.ensicaen.pi.gpss.backend.database.dto.TemplateDTO;
import fr.ensicaen.pi.gpss.backend.database.dto.beneficiary.BeneficiaryToBankAccountDTO;
import fr.ensicaen.pi.gpss.backend.database.dto.direct_debit.DirectDebitToBankAccountDTO;
import fr.ensicaen.pi.gpss.backend.database.dto.payment_method.PaymentMethodManagerDTO;
import fr.ensicaen.pi.gpss.backend.database.dto.request_payment_method.RequestPaymentMethodManagerDTO;
import fr.ensicaen.pi.gpss.backend.database.dto.transaction.TransactionToBankAccountDTO;
import fr.ensicaen.pi.gpss.backend.database.enumerate.BankAccountType;
import jakarta.validation.Valid;
import jakarta.validation.constraints.PositiveOrZero;
import lombok.Builder;
import lombok.Getter;

import java.util.List;

@Builder
@Getter
@JsonInclude(value = JsonInclude.Include.NON_EMPTY, content = JsonInclude.Include.NON_NULL)
@Valid
public class BankAccountDTO extends TemplateDTO {
    @PositiveOrZero
    private Integer idBankAccount;
    private String iban;
    private String rib;
    @PositiveOrZero
    private Integer amount;
    private BankAccountType bankAccountType;
    private List<TransactionToBankAccountDTO> transactions;
    private List<RequestPaymentMethodManagerDTO> paymentMethodRequests;
    private List<PaymentMethodManagerDTO> paymentMethods;
    private List<BeneficiaryToBankAccountDTO> beneficiaries;
    private List<DirectDebitToBankAccountDTO> directDebits;
}

File: Front-End/src/app/DTO/bank_account/BankAccount.dto.ts
import {PaymentMethodRequestDTO} from "../request_payment_method/PaymentMethodRequest.dto";
import {PaymentMethodDTO} from "../payment_method/PaymentMethod.dto";
import {TransactionToBankAccountDTO} from "../transaction/TransactionToBankAccount.dto";
import {DirectDebitToBankAccountDTO} from "../direct_debit/DirectDebitToBankAccount.dto";
import {BeneficiaryToBankAccountDTO} from "../beneficiary/BeneficiaryToBankAccount.dto";

export interface BankAccountDTO {
    amount: number;
    iban: string;
    directDebits: DirectDebitToBankAccountDTO[];
    bankAccountType: string;
    paymentMethodRequests: PaymentMethodRequestDTO[];
    paymentMethods: PaymentMethodDTO[];
    rib: string;
    idBankAccount: number;
    transactions: TransactionToBankAccountDTO[];
    beneficiaries: BeneficiaryToBankAccountDTO[];
}
